src/index.ts

```diff
@@ -1,3 +1,166 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 'use-strict';
+
+import {
+  IInputAreaViewModel
+} from 'jupyter-js-input-area';
+
+import {
+  IOutputAreaViewModel
+} from 'jupyter-js-output-area';
+
+import {
+  IObservableList
+} from 'phosphor-observablelist';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+
+/**
+ * An enum which describes the type of cell.
+ */
+enum CellType {
+  /**
+   * The cell contains code input.
+   */
+  Code,
+
+  /**
+   * The cell contains markdown.
+   */
+  Markdown,
+
+  /**
+   * The cell contains raw text.
+   */
+  Raw
+}
+
+
+/**
+ * The arguments object emitted with the `stateChanged` signal.
+ */
+export
+interface ICellChangedArgs<T> {
+  name: string,
+  oldValue: T;
+  newValue: T;
+}
+
+
+/**
+ * The definition of a model object for a base cell.
+ */
+interface IBaseCellViewModel {
+
+  /**
+   * The type of cell.
+   */
+  cellType: CellType;
+
+  /**
+   * Tags applied to the cell.
+   */
+  tags?: IObservableList<string>;
```

Comment:
Fine with `type`.

For the metadat stuff, what about making the values be something like this:

```
interface ISerializable {
  toJson(): any;
  fromJson(data: any): void;
}
```

That at least encodes the fact that we expect the data to be serializable
to/from JSON?



On Tue, Nov 3, 2015 at 10:20 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> In src/index.ts
> <https://github.com/jupyter/jupyter-js-cells/pull/2#discussion_r43839214>:
>
> > +
> > +
> > +/**
> > + * The definition of a model object for a base cell.
> > + */
> > +interface IBaseCellViewModel {
> > +
> > +  /**
> > +   * The type of cell.
> > +   */
> > +  cellType: CellType;
> > +
> > +  /**
> > +   * Tags applied to the cell.
> > +   */
> > +  tags?: IObservableList<string>;
>
> Never mind, I confusing with JSON schema *paternProperties*.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-cells/pull/2/files#r43839214>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com